crates/core/src/backend/decrypt.rs

```diff
@@ -395,12 +411,41 @@ impl<C: CryptoKey> DecryptWriteBackend for DecryptBackend<C> {
             }
             None => self.key().encrypt_data(data)?,
         };
-        let id = hash(&data);
-        self.write_bytes(tpe, &id, false, data.into())
+        if self.extra_check {
+            let check_data = self.decrypt_file(&data_encrypted)?;
```

Comment:
We might also want to implement a test on our side, like `data == decrypt(encrypt(data))` with different compression levels, just to be on the safe side for the future, that we catch possible regressions in that regard. I know, the library itself should actually do this and it's not our thing. But as we are relying here on this kind of existential feature of another crate, it might be good, to also secure the integrity on our side? 🤔 Not sure though.

Reply:
No - any test we find useful here, should instead be added to the library. And I think there are more competent people to decide on what tests are useful..